api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -344,7 +344,8 @@ public void changeQuestionAnswer(User u, String question, String answer) throws
 		
 		LoginCredential credentials = getLoginCredential(u);
 		credentials.setSecretQuestion(question);
-		credentials.setSecretAnswer(answer);
+		String hashedAnswer = Security.encodeString(answer + credentials.getSalt());

```

Comment:
I would suggest `answer.toLowerCase()` to make secret answer validation case insensitive.